examples/shop/src/utils.ts

```diff
@@ -0,0 +1,11 @@
+export function calculateCircularIndex(index: number, arrayLength: number) {
```

Comment:
This true circular function does the same:
``` ts
const calculateCircularIndex = (index: number, arrayLength: number) =>  (arrayLength + index) % arrayLength 
```

But it also supports this:

``` ts
setProductIndex(calculateCircularIndex(productIndex - 2, products.length));
setProductIndex(calculateCircularIndex(productIndex - 3, products.length));
setProductIndex(calculateCircularIndex(productIndex + 4, products.length));
```

Reply:
fixed